contracts/modules/staking/PooledStaking.sol

```diff
@@ -126,6 +128,15 @@ contract PooledStaking is MasterAware, IPooledStaking {
   uint public processedToStakerIndex; // we processed the action up this staker
   bool public isContractStakeCalculated; // flag to indicate whether staked amount is up to date or not
 
+  /* state vars for rewards groupping upgrade */
+
+  // rewards to be distributed at the end of the current round
+  // contract address => ContractRewards
+  mapping(address => ContractReward) public accumulatedRewards;
+
+  uint public REWARD_ROUND_DURATION;
+  uint public REWARD_ROUNDS_START;
+
   /* Modifiers */
 
   modifier noPendingActions {
```

Comment:
in regards to noPending modifiers:

noPendingUnstakeRequests and noPendingRewards seem to be unused now, can be removed

also noPendingActions and noPendingBurns are used only once, could inline the require there, either way it's fine